tik/lib/Util.cpp

```diff
@@ -13,24 +14,91 @@ using namespace llvm;
 
 namespace TraceAtlas::tik
 {
-
     std::map<int64_t, llvm::BasicBlock *> IDToBlock;
     std::map<int64_t, llvm::Value *> IDToValue;
+    void RecurseThroughOperands(Value *val)
+    {
+        if (auto inst = dyn_cast<Instruction>(val))
+        {
+            if (GetValueID(inst) < 0)
+            {
+                throw AtlasException("Found an instruction that did not have a ValueID.");
+            }
+            if (IDToValue.find(GetValueID(val)) == IDToValue.end())
+            {
+                IDToValue[GetValueID(val)] = val;
+            }
+            else
+            {
+                return;
+            }
+            for (unsigned int i = 0; i < inst->getNumOperands(); i++)
+            {
+                if (auto subVal = dyn_cast<Value>(inst->getOperand(i)))
+                {
+                    RecurseThroughOperands(subVal);
+                }
+            }
+        }
+        else if (auto go = dyn_cast<GlobalObject>(val))
+        {
+            if (GetValueID(go) < 0)
+            {
+                PrintVal(go);
```

Comment:
Should only be used for debugging. If this is ready to merge it should be removed.

Reply:
fixed